lib/bandit/telemetry.ex

```diff
@@ -76,8 +76,8 @@ defmodule Bandit.Telemetry do
       * `conn`: The `Plug.Conn` representing this connection. Not present in cases where `error`
         is also set and the nature of error is such that Bandit was unable to successfully build
         the conn
-      * `kind`: The kind of unexpected condition, typically `:exit`
-      * `exception`: The exception which caused this unexpected termination
+      * `kind`: The kind of unexpected condition, typically `:error` or `:throw`
+      * `reason`: The error reason. An Exception struct or the thrown value
```

Comment:
The rest of this PR looks great; these are the only two items I have issue with.

For better or worse, we've already baked in `exception` and `exit` in metadata, and [downstream packages](https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/instrumentation/opentelemetry_bandit/lib/opentelemetry_bandit.ex) already depend on them. We can't change them without a major version bump and that's not happening for this. 

I'd suggest:

* Changing `:error` back to `:exit` and `reason` back to `exception`
* Keeping `:throw` as a second valid value for`kind`
* Adding a new `:throw_value` key to pass the thrown value if it's a throw

This is enough to be backwards compatible and still have sensible semantic here I think.

Reply:
Thanks for the feedback!

Sounds fair.

Related to this previous discussion, for what is worth, https://github.com/mtrudel/bandit/pull/398#discussion_r1761940338.

Will try adapt to something backwards-compatible.

Reply:
After digging into `opentelemetry_bandit` I realized it might be a problem to start emitting events of `kind: :throw` right now. `opentelemetry_bandit` (actually `opentelemetry_api`) is expecting only exceptions: https://github.com/open-telemetry/opentelemetry-erlang/blob/main/apps/opentelemetry_api/lib/open_telemetry/span.ex.

The story is different for cowboy because it's using `otel_span.erl`, which actually handles gracefully different kinds (referred to as `Class` in there): https://github.com/open-telemetry/opentelemetry-erlang/blob/a506d08e8439c353bbf8277a1ca619c668d125f8/apps/opentelemetry_api/src/otel_span.erl#L239-L242. 

Rolling back for now, to only emitting the same events as before this PR and later can tackle emitting new throw events in tandem with possible updates in `opentelemetry_bandit`.

Reply:
> Rolling back for now, to only emitting the same events as before this PR and later can tackle emitting new throw events in tandem with possible updates in `opentelemetry_bandit`.

https://github.com/mtrudel/bandit/pull/411/commits/e8239b612a8003f05effe657e862aa37b451b412